src/Halogen/Store/Connect.purs

```diff
@@ -73,6 +73,7 @@ connect (Selector selector) component =
 
   handleAction = case _ of
     Initialize -> do
+      (Selector selector) <- mkSelector <$> H.gets _.input
       subscribe (Selector selector) Update
```

Comment:
There's a problem here, the selector will be constructed with the input the component had at the time of `Initialize` and that gets passed to subscribe, so all the subsequent times the selector is run it will still get the initial input.

One solution for this that I thought is to add a variation of `emitSelected` that would look something like this:

```purescript
  emitSelectedWithInput :: forall input s'. m input -> (input -> Selector s s') -> m (Emitter s')
```

So `H.gets _.input` would be passed to it as a way to get the current input every time the emitter runs.

But I'm not sure if this is the right way to go. Wdyt?

As a side note, for the use case I had the stale input wouldn't be a problem. Basically I have a component that renders a list of components, and I pass the `id` of each element to the corresponding child component. The child component then would be able to select their particular element from the list instead of having to select the whole list. So re-renders would happen only when that particular element changed, instead of the whole list.

Reply:
Yes, I agree that the selector needs to be able to get the current input each time the emitter runs. And it would be possible to add a method to the `MonadStore` class as you described (`emitSelectedWithInput`). I think this is a reasonable approach (the alternative would be to end the current subscription and start a new subscription any time the input changes).

Reply:
I tried both implementations.

The re-subscription one is a bit messy and could have some concurrency issues but it works (or at least it compiles :sweat_smile:).

I could not make the option of passing down `m input` work, because the action inside `StoreT` run in a different `m` (in the implementation of `emitSelected`) than the one I'm passing down.

However I came up with another idea that simplifies things, instead of applying the input to the selector we could pass a function to `connect` that would filter the output of the selector based on the input

```purescript
connect
  :: forall action store contextA contextB query input output m
   . MonadEffect m
  => Eq contextB
  => MonadStore action store m
  => (input -> contextA -> contextB)
  -> Selector store contextA
  -> H.Component query (Connected contextB input) output m
  -> H.Component query input output m
```

We then can apply the input and the selector's context to `input -> contextA -> contextB` and get the context that the connected component will receive.

However I had to include an `Eq` constraint to avoid `unsafeRefEq` in this line:

```purescript
Just oldContext | eq oldContext newContext -> pure unit
```

Now, after all of this, I realized that this same optimization could be done on user land by filtering the context of the selector in the `Receive` function. And there one would have access not only to the component's input but also to it's state.

So I don't think this PR is needed after all. Unless you think my last idea from above of _post processing_ the selector's output makes sense to have. Then I can push the changes I have.